src/pages/test-report/components/toolbar/toolbar.ts

```diff
@@ -50,7 +57,14 @@ export class ToolbarComponent {
   }
 
   toggleSeriousMode(): void {
+    if (this.isDangerousMode) {
+      this.store$.dispatch(new ToggleDangerousFaultMode());
+    }
     this.store$.dispatch(new ToggleSeriousFaultMode());
   }
 
+  toggleDangerousMode(): void {
+    this.store$.dispatch(new ToggleDangerousFaultMode());
```

Comment:
Shouldn't you toggle the serious mode if it's on?

Reply:
Made the change :)